gdal/frmts/gtiff/geotiff.cpp

```diff
@@ -12541,6 +12542,20 @@ GDALDataset *GTiffDataset::Open( GDALOpenInfo * poOpenInfo )
     if( STARTS_WITH_CI(pszFilename, "GTIFF_DIR:") )
         return OpenDir( poOpenInfo );
 
+    int nDatasetId = 0;
+    if (STARTS_WITH_CI(pszFilename, "GTIFF_DATASET:"))
```

Comment:
I'm not clear on the semantic difference between GTIFF_DIR: and GTIFF_DATASET:
With the SubIFD mechanism, they should be identical.

Reply:
GTIFF_DIR goes through OepnDir() instead of Open() and thus opens the dataset with m_bBase=false, which in turn actually prevents the directory scan from finding associated overviews. I can change the code to set m_bBase=true and adapt ScanDirectories accordingly if you prefer, I'm just not certain if this can have unwanted side-effects.

Reply:
Or perhaps use GTIFF_DIR, add a m_bIsOpenOnSubDir = true, and modify ScanDirectories()  to only look at the SubIFD of the current directory when m_bIsOpenOnSubDir = true, and not iterate on top-level IFD.

I believe we should also get rid of the m_bBase flag completely:
- remove the if( m_bBase ) test in GTiffDataset::Finalize(). It is probably not needed currently
- remove it in  GTiffDataset::CloseDependentDatasets() as well
- and completely remove the few remaining uses of m_bBase by checking m_poBaseDS instead. Basically when currently we have m_bBase == true, this is equivalent to m_poBaseDS == nullptr. m_poBaseDS is a pointer from the mask or overview to the base dataset, and it is null for the base dataset itself